code/game/objects/items/weapons/implants/implant_stealth.dm

```diff
@@ -0,0 +1,161 @@
+	INVOKE_ASYNC(box, TYPE_PROC_REF(/obj/structure/closet/cardboard/agent, go_invisible), 1.7 SECONDS)
+	box.create_fake_box()
+	owner.forceMove(box)
+
+/datum/action/item_action/agent_box/proc/recall_box_animation()
+	var/image/fake_box = image('icons/obj/cardboard_boxes.dmi', owner, "agentbox", ABOVE_MOB_LAYER)
+	flick_overlay_view(fake_box, owner, 0.4 SECONDS)
+	animate(fake_box, pixel_z = fake_box.pixel_z + 30, alpha = fake_box.alpha - 255, time = 3, loop = 1)
+
+/datum/action/item_action/agent_box/Grant(mob/grant_to)
+	. = ..()
+	if(owner)
+		RegisterSignal(owner, COMSIG_HUMAN_SUICIDE_ACT, PROC_REF(suicide_act))
+
+/datum/action/item_action/agent_box/Remove(mob/M)
+	if(owner)
+		UnregisterSignal(owner, COMSIG_HUMAN_SUICIDE_ACT)
+	return ..()
+
+/datum/action/item_action/agent_box/proc/suicide_act(datum/source)
+	SIGNAL_HANDLER
+
+	if(!istype(owner.loc, /obj/structure/closet/cardboard/agent))
+		return
+	var/obj/structure/closet/cardboard/agent/box = owner.loc
+	owner.visible_message("<span class='danger'>[owner] falls out of [box]! It looks like [owner.p_they()] committed suicide!</span>")
+	owner.playsound_local(box, 'sound/misc/box_deploy.ogg', 50, TRUE)
+	INVOKE_ASYNC(box, TYPE_PROC_REF(/obj/structure/closet/cardboard/agent, open))
+	INVOKE_ASYNC(owner, TYPE_PROC_REF(/atom/movable, throw_at), get_turf(owner))
+	return OXYLOSS
+
+// Stealth implant box
+/obj/structure/closet/cardboard/agent
+	name = "inconspicious box"
+	desc = "It's so normal that you didn't notice it before."
+	icon_state = "agentbox"
+	max_integrity = 1
+	move_speed_multiplier = 0.5 // You can move at normal walk speed while in this box.
+	/// UID of the person who summoned this box with an implant.
+	var/implant_user_UID
+	// This has to be a separate object and not just an image because the image will inherit the box's 0 alpha while it is stealthed.
+	/// A holder effect which follows the src box so we can display an image to the person inside the box.
+	var/obj/effect/fake_box
+	/// The box image attached to the `fake_box` object.
+	var/image/box_img
+
+/obj/structure/closet/cardboard/agent/Destroy()
+	if(implant_user_UID)
+		var/mob/living/implant_user = locateUID(implant_user_UID)
+		implant_user.client?.images -= box_img
+	QDEL_NULL(fake_box)
+	QDEL_NULL(box_img)
+	return ..()
+
+/obj/structure/closet/cardboard/agent/open()
+	. = ..()
+	if(!.)
+		return FALSE
+	qdel(src)
+
+// When the box is opened, it's deleted, so we never need to update this.
+/obj/structure/closet/cardboard/agent/update_icon_state()
+	return
+
+/obj/structure/closet/cardboard/agent/proc/create_fake_box()
+	if(fake_box)
+		return
+	fake_box = new(get_turf(src))
+	fake_box.mouse_opacity = MOUSE_OPACITY_TRANSPARENT // This object should be completely invisible.
+	box_img = image(icon, fake_box, icon_state, ABOVE_MOB_LAYER)
+	box_img.alpha = 128
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(move_fake_box))
+	var/mob/living/implant_user = locateUID(implant_user_UID)
```

Comment:
is there any situation where resolving this UID will produce a null value due to a qdel?

Reply:
I suppose it's possible but it would be a very tight window. They would have to be basically in the process of being deleted immediately _after_ they've pressed their action button.